packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,39 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldNotContains = target !== this && !this.contains(relatedTarget as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // keep current page if tabbing back into the carousel after previously moving pages
+    if (currentContains && oldNotContains && currentCardIndex === 0) {
+      (this.children[this.start] as HTMLElement).focus();
+      return;
+    }
+
+    if (currentContains) {
+      // going forwards, change page depending on card index
+      if (currentCardIndex >= this.start + this.pageSize) {
+        const nextStart = currentCardIndex - (currentCardIndex % this.pageSize);
+        const pageOffset = this.start % this.pageSize;
+
+        this.start = nextStart + pageOffset;
+
+        // going backwards, change page depending on card index
+      } else if (currentCardIndex < this.start) {
+        this.start = Math.max(currentCardIndex + 1 - this.pageSize, 0);
```

Comment:
Do you want to apply your improvement to "page forward" logic to the "page backward" logic here, too?

Reply:
Since we'd use the same code going in either direction, the previous logic runs in both directions whenever the focus is out of the current page.